src/EditModel/EditDataEntryForm/processFormData.js

```diff
@@ -0,0 +1,102 @@
+import log from 'loglevel';
+
+// TODO: write tests for processFormData
+
+export const generateHtmlForId = (id, {
+    styleAttr,
+    disabledAttr,
+    insertGrey,
+    operands,
+    totals,
+    indicators,
+}) => {
+    const style = styleAttr ? ` style=${styleAttr}` : '';
+    const disabled = disabledAttr || insertGrey ? ' disabled="disabled"' : '';
+
+    if (id.includes('-')) {
+        const label = operands && operands[id];
+        const attr = `name="entryfield" title="${label}" value="[ ${label} ]"${style}${disabled}`.trim();
+        return `<input id="${id}" ${attr}/>`;
+    } else if (totals.hasOwnProperty(id)) {
+        const label = totals[id];
+        // TODO: `name="total"` is included twice - can we remove one of them or
+        // would this break something?
+        const attr = `name="total" readonly title="${label}" value="[ ${label} ]"${style}${disabled}`.trim();
+        return `<input dataelementid="${id}" id="total${id}" name="total" ${attr}/>`;
+    } else if (indicators.hasOwnProperty(id)) {
+        const label = indicators[id];
+        const attr = `name="indicator" readonly title="${label}" value="[ ${label} ]"${style}${disabled}`.trim();
+        return `<input indicatorid="${id}" id="indicator${id}" ${attr}/>`;
+    }
+
+    log.warn('Failed to generate HTML for ID:', id);
+    return '';
+}
+
+
+const dataElementCategoryOptionIdPattern = /id="(\w*?)-(\w*?)-val"/;
+const dataElementPattern = /dataelementid="(\w{11})"/;
+const indicatorPattern = /indicatorid="(\w{11})"/;
+
+const getIdFromInputHtml = inputHtml => {
+    const idMatch = dataElementCategoryOptionIdPattern.exec(inputHtml);
+    const dataElementTotalMatch = dataElementPattern.exec(inputHtml);
+    const indicatorMatch = indicatorPattern.exec(inputHtml);
+
+    if (idMatch) {
+        return `${idMatch[1]}-${idMatch[2]}-val`;
+    } else if (dataElementTotalMatch) {
+        return dataElementTotalMatch[1];
+    } else if (indicatorMatch) {
+        return indicatorMatch[1];
+    }
+}
+
+const inputPattern = /<input.*?\/>/gi;
+
+export const processFormData = ({
```

Comment:
Extracted from https://github.com/dhis2/maintenance-app/blob/224f2ec40242776d1e92ab5d6fdc2cc9d4989c88/src/EditModel/EditDataEntryForm.component.js#L320-L362

I refactored the ID matching into `getIdFromInputHtml`